src/check/check_result.ts

```diff
@@ -0,0 +1,54 @@
+import { formatCheckResult, getRPCUrl } from "../utils";
+import { OracleFetchOptions, OracleResponse } from "../types";
+
+// TODO - Fix this/change this, maybe update the type to reduce number of checks
+function _getMaxIterations(opts: OracleFetchOptions) {
+  if (opts.network === "auto") {
+    if (
+      opts.nameOrRpcUrl.includes("staging") ||
+      opts.nameOrRpcUrl.includes("testnet")
+    ) {
+      return 3;
```

Comment:
Let's add these values as constants like DEFAULT_MAX_ITERATIONS_TESTNET and mainnet

and let's add optional param "max iterarions" to fetch options - to be able to customize it

Reply:
Agree - will push update